ownership-chain/e2e-tests/tests/test-evolution.ts

```diff
@@ -0,0 +1,164 @@
+import { describeWithExistingNode, slotAndOwnerToTokenId } from "./util";
+import { step } from "mocha-steps";
+import { CONTRACT_ADDRESS, GAS, GAS_PRICE, GENESIS_ACCOUNT, GENESIS_ACCOUNT_PRIVATE_KEY, LAOS_EVOLUTION_ABI, SELECTOR_LOG_EVOLVED_WITH_EXTERNAL_TOKEN_URI, SELECTOR_LOG_MINTED_WITH_EXTERNAL_TOKEN_URI, SELECTOR_LOG_NEW_COLLECTION } from "./config";
+import { expect } from "chai";
+import Contract from "web3-eth-contract";
+import BN from "bn.js";
+
+describeWithExistingNode("Frontier RPC (Mint and Evolve Assets)", (context) => {
+
+    let contract: Contract;
+    let nonce: number;
+    let collectionId: number = 0;
+
+    beforeEach(async function () {
+        this.timeout(70000);
+
+        contract = new context.web3.eth.Contract(LAOS_EVOLUTION_ABI, CONTRACT_ADDRESS, {
+            from: GENESIS_ACCOUNT,
+            gasPrice: GAS_PRICE,
+        });
+
+        nonce = await context.web3.eth.getTransactionCount(GENESIS_ACCOUNT);
+
+        context.web3.eth.accounts.wallet.add(GENESIS_ACCOUNT_PRIVATE_KEY);
+
+        const result = await contract.methods.createCollection(GENESIS_ACCOUNT).send({ from: GENESIS_ACCOUNT, gas: GAS, nonce: nonce++ });
+        expect(result.status).to.be.eq(true);
+        collectionId = result.events.NewCollection.returnValues.collectionId;
+    });
+
+    step("when collection does not exist token uri should fail", async function () {
+
+        const tokenId = "0";
+
+        try {
+            await contract.methods.tokenURI(collectionId, tokenId).call();
+            expect.fail("Expected error was not thrown"); // Ensure an error is thrown
+        } catch (error) {
+            expect(error.message).to.be.eq(
+                "Returned error: VM Exception while processing transaction: revert"
+            );
+        }
+    });
+
+    step("when asset is minted it should return token uri", async function () {
+        this.timeout(70000);
+
+        const slot = "0";
+        const to = GENESIS_ACCOUNT;
+        const tokenURI = "https://example.com";
+
+        const result = await contract.methods.mintWithExternalURI(collectionId, slot, to, tokenURI).send({ from: GENESIS_ACCOUNT, gas: GAS, nonce: nonce++ });
+        expect(result.status).to.be.eq(true);
+
+        const tokenId = result.events.MintedWithExternalURI.returnValues.tokenId;
+        const got = await contract.methods.tokenURI(collectionId, tokenId).call();
+        expect(got).to.be.eq(tokenURI);
+    });
+
+    step("given slot and owner it should return token id", async function () {
+        this.timeout(70000);
+
+        const slot = "1";
+        const to = GENESIS_ACCOUNT;
+
+        const tokenId = slotAndOwnerToTokenId(slot, to);
+        expect(tokenId).to.be.eq("000000000000000000000001c0f0f4ab324c46e55d02d0033343b4be8a55532d");
+        const tokenIdDecimal = new BN(tokenId, 16).toString(10);
```

Comment:
is this `LE` or `BE` by default?

Reply:
btw, `mintWithExternalUri` can also be extracted into a separate function, since it's used couple of times as well

Reply:
https://github.com/freeverseio/laos/pull/138#discussion_r1382965737

Good point. I don't see where the library specify any default value so I assume by default it takes platform's native endianness. But I'll specify so we avoid unexpected behaviours